File: .github/Wallet.cs
using System;
using System.Collections.Generic;

public class Wallet
{
    private double money;
    private double saving;
    private List<(string type, double amount)> transactions;

    public Wallet()
    {
        money = 0;
        saving = 0;
        transactions = new List<(string type, double amount)>();
    }

    public void Deposit(double amount)
    {
        money += amount;
        transactions.Add(("Deposit", amount));
        Console.WriteLine($"Deposit of {amount} done. New balance: {money} Ariary");
    }

    public void Withdraw(double amount)
    {
        if (money >= amount)
        {
            money -= amount;
            transactions.Add(("Withdrawal", amount));
            Console.WriteLine($"Withdrawal of {amount} done. New balance: {money} Ariary");
        }
        else
        {
            Console.WriteLine("Insufficient funds to make the withdrawal");
        }
    }

    public void SavingOperation(double amount)
    {
        if (money >= amount)
        {
            money -= amount;
            saving += amount;
            transactions.Add(("Saving", amount));
            Console.WriteLine($"Saving of {amount} done. New balance: {money} Ariary");
        }
        else
        {
            Console.WriteLine("Insufficient funds to make the saving");
        }
    }

    public void TransactionHistory()
    {
        Console.WriteLine("\nTransaction history: ");
        foreach (var transaction in transactions)
        {
            Console.WriteLine($"{transaction.type} of {transaction.amount}");
        }
        Console.WriteLine($"Final balance: {money} Ariary");
        Console.WriteLine($"Saved money: {saving} Ariary");
    }

    public void CurrencyConversion(double currency, double amount)
    {
        double amountConverted = 0;
        if (currency == 1)
        {
            amountConverted = amount / 4582;
            Console.WriteLine($"{amount} Ariary in dollars: {amountConverted} $");
        }
        else if (currency == 2)
        {
            amountConverted = amount / 4933;
            Console.WriteLine($"{amount} Ariary in Euro: {amountConverted} €");
        }
        else if (currency == 3)
        {
            amountConverted = amount / 31;
            Console.WriteLine($"{amount} Ariary in Yen: {amountConverted} ¥");
        }
    }

    public void Operations()
    {
        bool exit = false;

        while (!exit)
        {
            Console.WriteLine("\nChoose the operation to perform: \n1-Deposit \n2-Withdrawal \n3-Save \n4-History \n5-Currency Conversion \n6-Exit");
            double option;
            if (double.TryParse(Console.ReadLine(), out option))
            {
                switch (option)
                {
                    case 1:
                        Console.WriteLine("Enter the amount to deposit: ");
                        double depositAmount;
                        if (double.TryParse(Console.ReadLine(), out depositAmount))
                        {
                            Deposit(depositAmount);
                        }
                        else
                        {
                            Console.WriteLine("Invalid input. Please enter a valid number.");
                        }
                        break;
                    case 2:
                        Console.WriteLine("Enter the amount to withdraw: ");
                        double withdrawalAmount;
                        if (double.TryParse(Console.ReadLine(), out withdrawalAmount))
                        {
                            Withdraw(withdrawalAmount);
                        }
                        else
                        {
                            Console.WriteLine("Invalid input. Please enter a valid number.");
                        }
                        break;
                    case 3:
                        Console.WriteLine("Enter the amount to save: ");
                        double savingAmount;
                        if (double.TryParse(Console.ReadLine(), out savingAmount))
                        {
                            SavingOperation(savingAmount);
                        }
                        else
                        {
                            Console.WriteLine("Invalid input. Please enter a valid number.");
                        }
                        break;
                    case 4:
                        TransactionHistory();
                        break;
                    case 5:
                        Console.WriteLine("Choose the currency: \n1-Dollar \n2-Euro \n3-Yen ");
                        double currencyType;
                        if (double.TryParse(Console.ReadLine(), out currencyType))
                        {
                            Console.WriteLine("Enter the amount to convert: ");
                            double conversionAmount;
                            if (double.TryParse(Console.ReadLine(), out conversionAmount))
                            {
                                CurrencyConversion(currencyType, conversionAmount);
                            }
                            else
                            {
                                Console.WriteLine("Invalid input. Please enter a valid number.");
                            }
                        }
                        else
                        {
                            Console.WriteLine("Invalid input. Please enter a valid number.");
                        }
                        break;
                    case 6:
                        exit = true;
                        break;
                    default:
                        Console.WriteLine("Invalid choice. Please choose a valid option.");
                        break;
                }
            }
            else
            {
                Console.WriteLine("Invalid input. Please enter a valid number.");
            }
        }
    }
}

class Program
{
    static void Main()
    {
        Wallet wallet = new Wallet();
        wallet.Operations();
    }
}

File: .github/Wallet.py
class Wallet:
    def __init__(self):
        self.money = 0
        self.saving = 0
        self.transactions = []

    def deposit(self, amount):
        self.money += amount
        self.transactions.append(("Deposit", amount))
        print(f"Deposit of {amount} done. New balance: {self.money} Ariary")

    def withdraw(self, amount):
        if self.money >= amount:
            self.money -= amount
            self.transactions.append(("Withdrawal", amount))
            print(f"Withdrawal of {amount} done. New balance: {self.money} Ariary")
        else:
            print("Insufficient funds to make the withdrawal")

    def saving_operation(self, amount):
        if self.money >= amount:
            self.money -= amount
            self.saving += amount
            self.transactions.append(("Saving", amount))
            print(f"Saving of {amount} done. New balance: {self.money} Ariary")
        else:
            print("Insufficient funds to make the saving")

    def transaction_history(self):
        print("\nTransaction history: ")
        for transaction in self.transactions:
            print(f"{transaction[0]} of {transaction[1]}")
        print(f"Final balance: {self.money} Ariary")
        print(f"Saved money: {self.saving} Ariary")

    def currency_conversion(self, currency, amount):
        amount_converted = 0
        if currency == 1:
            amount_converted = amount / 4582
            print(f"{amount} Ariary in dollars: {amount_converted} $")
        elif currency == 2:
            amount_converted = amount / 4933
            print(f"{amount} Ariary in Euro: {amount_converted} €")
        elif currency == 3:
            amount_converted = amount / 31
            print(f"{amount} Ariary in Yen: {amount_converted} ¥")

    def operations(self):
        exit_flag = False
        while not exit_flag:
            print("\nChoose the operation to perform: \n1-Deposit \n2-Withdrawal \n3-Save \n4-History \n5-Currency Conversion \n6-Exit")
            option = int(input("Enter your choice: "))
            if option == 1:
                deposit_amount = float(input("Enter the amount to deposit: "))
                self.deposit(deposit_amount)
            elif option == 2:
                withdrawal_amount = float(input("Enter the amount to withdraw: "))
                self.withdraw(withdrawal_amount)
            elif option == 3:
                saving_amount = float(input("Enter the amount to save: "))
                self.saving_operation(saving_amount)
            elif option == 4:
                self.transaction_history()
            elif option == 5:
                currency_type = int(input("Choose the currency: \n1-Dollar \n2-Euro \n3-Yen "))
                conversion_amount = float(input("Enter the amount to convert: "))
                self.currency_conversion(currency_type, conversion_amount)
            elif option == 6:
                exit_flag = True
            else:
                print("Invalid choice. Please choose a valid option.")


wallet = Wallet()
wallet.operations()